lib/avo/base_resource.rb

```diff
@@ -151,7 +151,7 @@ def hydrate(model: nil, view: nil, user: nil, params: nil)
       @user = user if user.present?
       @params = params if params.present?
 
-      if model.present?
+      if model.present? && model.class == model_class
```

Comment:
I'm not sure this is the proper place to fix this. I mean, `BaseResource.model_class` falls back to computing the model class based on it's own name.
It seems a bit arbitrary.

How about finding the place where we wrongly hydrate that resource and applying the fix there?

Reply:
Hmm, the issue starts here:

![image](https://user-images.githubusercontent.com/69730720/203822407-a9b0fef6-5973-401c-8e71-6a0efc25ef6c.png)

I'm not sure how we can verify it without model_class... Yes, mostly it's computed but sometimes it's not, and i think this covers both situations. It's there any situation where we want a resource to have a model different from his model_class, ancestors or dependents?

Reply:
What I mean is, fix the issue where we run `resource.hydrate(model: WRONG_MODEL)` instead of checking inside the resource. Because it's clear to me that somewhere we're making that mistake.

Reply:
Got it, i agree, issue found: 

![image](https://user-images.githubusercontent.com/69730720/203826939-9c6afc72-98a4-4f40-846b-993d9d2dda93.png)